scout: add sonarqube integration (#18725)
docker/docs: content/scout/integrations/code-quality/sonarqube.md

The SonarQube integration enables Docker Scout to evaluate the [Quality gates
passed policy](../../policy/_index.md#quality-gates-passed) for a repository,
using analysis results from SonarQube quality gates.

## How it works

This integration uses [SonarQube
webhooks](https://docs.sonarsource.com/sonarqube/latest/project-administration/webhooks/)
to notify Docker Scout of when a SonarQube project analysis has completed. When
the webhook is called, Docker Scout receives the analysis results, and stores
them in the database.

When you trigger a policy evaluation for the image repository, Docker Scout
evaluates the results of the SonarQube analysis record that corresponds to the
image. Docker Scout uses Git provenance attestations on the image to link the
image repository with SonarQube analysis results.

Docker Scout doesn't have access to historic SonarQube analysis records. Only
analysis results recorded after the integration was enabled are available to
Docker Scout.

## Prerequisites

- Your image repository is [integrated with Docker Scout](../_index.md#container-registries)
- Your images are built with [provenance attestations](../../../build/attestations/slsa-provenance.md)

>   with information about the Git repository

Author: Presumably the `org.opencontainers.image.source` label is needed as well right?

>   with information about the Git repository

Reviewer: I don't think so. Scout uses the image sha to link the SonarQube analysis.

## Enable the SonarQube integration

1. Go to [Integrations](https://scout.docker.com/settings/integrations/) on the
   Docker Scout Dashboard.
2. Select the **See all integration options** filter.
3. Find the SonarQube integration in the **Code quality gates** section, and

>    select **Integrate**.

Author: I wish :'-) we use select instead of click.

---
title: Integrate Docker Scout with SonarQube
description: Evaluate your images with the SonarQube quality gates defined in your projects
keywords: scout, supply chain, integration, code quality
---

The SonarQube integration enables Docker Scout to surface SonarQube quality
gate checks through Policy Evaluation, under a new [Quality gates passed
policy](../../policy/_index.md#quality-gates-passed).

## How it works

This integration uses [SonarQube
webhooks](https://docs.sonarsource.com/sonarqube/latest/project-administration/webhooks/)
to notify Docker Scout of when a SonarQube project analysis has completed. When
the webhook is called, Docker Scout receives the analysis results, and stores
them in the database.

When you push a new image to a repository, Docker Scout evaluates the results
of the SonarQube analysis record corresponding to the image. Docker Scout uses
Git provenance metadata on the images, from  provenance attestations or an OCI
annotations, to link image repositories with SonarQube analysis results.

> **Note**
>
> Docker Scout doesn't have access to historic SonarQube analysis records. Only

> > analysis results recorded after the integration was enabled are available to

Reviewer 2: Wondering if it reads better to rephrase it

```suggestion
> Docker Scout doesn't have access to historic SonarQube analysis records. Only
> analysis results recorded after the integration is enabled will be available to Docker Scout.
```

> Docker Scout.

## Prerequisites

> 

Reviewer 2: ```suggestion
To integrate Docker Scout with SonarQube, ensure that:
```

> - Your image repository is [integrated with Docker Scout](../_index.md#container-registries)

Reviewer 2: ```suggestion
- Your image repository is [integrated with Docker Scout](../_index.md#container-registries).
```
